session.go

```diff
@@ -259,158 +265,218 @@ func (sess *session) Break(c chan<- bool) {
 	sess.breakCh = c
 }
 
-func (sess *session) handleRequests(reqs <-chan *gossh.Request) {
-	for req := range reqs {
-		switch req.Type {
-		case "shell", "exec":
-			if sess.handled {
-				req.Reply(false, nil)
-				continue
-			}
+func (sess *session) handleRequests(ctx Context, reqs <-chan *gossh.Request) {
+	keepAlive := ctx.KeepAlive()
+	defer keepAlive.Close()
 
-			var payload = struct{ Value string }{}
-			gossh.Unmarshal(req.Payload, &payload)
-			sess.rawCmd = payload.Value
+	var keepAliveRequestInProgress sync.Mutex
+	for {
+		select {
+		case <-keepAlive.Ticks():
+			if keepAlive.TimeIsUp() {
+				log.Println("Keep-alive reply not received. Close down the session.")
 
-			// If there's a session policy callback, we need to confirm before
-			// accepting the session.
-			if sess.sessReqCb != nil && !sess.sessReqCb(sess, req.Type) {
-				sess.rawCmd = ""
-				req.Reply(false, nil)
-				continue
+				err := sess.Exit(0)
+				if err != nil {
+					log.Printf("Session exit failed: %v", err)
+				}
```

Comment:
I was looking at what exit code we should use (thinking 255 might be more appropriate), when I saw something gnarly: https://github.com/coder/ssh/blob/04bb837133e11ad897cef16572b3b732f76c4cc4/session.go#L193-L196

Essentially, if there's an error writing the message, the session will never be closed. Perhaps it's best to return to calling `Close` directly, like you were doing originally?

```suggestion
				_ = sess.Close()
```

Reply:
255 and `sess.Close` 👍  It could be an improvement, yes.

Reply:
I was thinking we would remove `sess.Exit` entirely, because when I thought about it I'm not sure we should be telling an "exit-status" to the client. The command that was running didn't exit, we just terminated the connection 🤔.